src/custom-elements.js

```diff
@@ -562,7 +573,7 @@ let Deferred;
           const node = walker.currentNode;
           if (node[_upgradedProp] && node[_attachedProp]) {
             node[_attachedProp] = false;
-            const definition = this._definitions.get(node.localName);
+            const definition = getDefinition(this._definitions, node);
```

Comment:
👍  I like that idea. Putting it on the element instead of the looking it up would improve performance. And getting rid of the `is` property will offset the new storage space for the definition.

And regarding the `is` property, just reviewed the spec and looks like I got confused between the ["is value"](https://dom.spec.whatwg.org/#concept-element-is-value) and an actual userland property on the dom element. So you're right, it should be removed from this polyfill